src/services/send-audit-events.ts

```diff
@@ -63,15 +77,53 @@ export async function sendAuditEvent(
  * Helper function to build extension object based on the type of event
  * @param event - Original event received from TxMA
  * @param eventEnum - Event name as an EventEnum
- * @param appliedAt - optional timestamp of when then event transition was applied
+ * @param finalState - Final state after intervention was/ was not applied
+ * @param txmaEventName - The name of the TxMA event name
  * @returns - TxMAEgressExtensions object
  */
-function buildExtensions(event: TxMAIngressEvent, eventEnum: EventsEnum, appliedAt?: number): TxMAEgressExtensions {
+function buildExtensions(
+  event: TxMAIngressEvent,
+  eventEnum: EventsEnum,
+  finalState: AccountStateEngineOutput,
+  txmaEventName: TxMAEgressEventName,
+): TxMAEgressExtensions {
   return {
-    eventType: userLedActionList.includes(eventEnum) ? 'USER_LED_ACTION' : 'TICF_ACCOUNT_INTERVENTION',
-    event: eventEnum,
+    trigger_event: event.event_name,
+    trigger_event_id: event.event_id ?? 'UNKNOWN',
+    description: userLedActionList.includes(eventEnum) ? 'USER_LED_ACTION' : finalState.interventionName!,
     intervention_code: event.extensions?.intervention?.intervention_code,
     reason: event.extensions?.intervention?.intervention_reason,
-    appliedAt,
+    allowable_interventions: finalState.nextAllowableInterventions.filter(
+      (intervention) => !nonInterventionsCodes.has(intervention),
+    ),
+    state: buildStateAttribute(finalState, eventEnum, txmaEventName),
+    action: buildActionAttribute(finalState, eventEnum),
   };
 }
+
+function buildStateAttribute(
+  finalState: AccountStateEngineOutput,
+  eventEnum: EventsEnum,
+  txmaEventName: TxMAEgressEventName,
+): State | undefined {
+  if (!finalState.finalState.suspended || (finalState.finalState.suspended && userLedActionList.includes(eventEnum)))
+    return State.ACTIVE;
+  if (finalState.finalState.suspended && !userLedActionList.includes(eventEnum)) return State.SUSPENDED;
+  if (finalState.finalState.blocked) return State.PERMANENTLY_SUSPENDED;
+  if (txmaEventName === 'AIS_EVENT_IGNORED_ACCOUNT_DELETED') return State.DELETED;
+}
+
+function buildActionAttribute(
```

Comment:
As an idea.. could this method be merged with the one above? The checks are really similar, and we only expect to have an action attribute in the message if the value of the state attribute is ACTIVE, so we could avoid doing those checks if that is not the case. You could then unpack a single returned object to get state and action attributes into the message object :)